unyt/array.py

```diff
@@ -1765,18 +1764,6 @@ def __pow__(self, p, mod=None, /):
         else:
             return super().__pow__(p, mod)
 
-    def __eq__(self, other):
-        try:
-            return super().__eq__(other)
-        except (IterableUnitCoercionError, UnitOperationError):
-            return np.zeros(self.shape, dtype="bool")
```

Comment:
have you tried running tests + coverage with numpy's bleeding edge version ? this code was supposed to support numpy 1.25 (yet to be released) (see https://github.com/yt-project/unyt/pull/333)